pkg/objstore/azure/azure.go

```diff
@@ -192,13 +195,16 @@ func (b *Bucket) GetRange(ctx context.Context, name string, off, length int64) (
 // Exists checks if the given object exists.
 func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
 	level.Debug(b.logger).Log("msg", "check if blob exists", "blob", name)
-	blobURL := getBlobURL(ctx, b.config.StorageAccountName, b.config.StorageAccountKey, b.config.ContainerName, name)
+	blobURL, err := getBlobURL(ctx, b.config.StorageAccountName, b.config.StorageAccountKey, b.config.ContainerName, name)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get Azure blob URL, address: %s", name)
+	}
 
-	if _, err := blobURL.GetProperties(ctx, blob.BlobAccessConditions{}); err != nil {
+	if _, err = blobURL.GetProperties(ctx, blob.BlobAccessConditions{}); err != nil {
 		if b.IsObjNotFoundErr(err) {
 			return false, nil
 		}
-		return false, errors.Wrapf(err, "cannot get blob URL: %s", name)
+		return false, errors.Wrapf(err, "cannot get Azure blob URL, address: %s", name)
```

Comment:
Can we wrap this with what we are actually doing? I think we are `getting properties` here not blob URL right?

Basically we want those two to be unique so we can easily track this down later